src/models/flight_control/FGLinearActuator.cpp

```diff
@@ -0,0 +1,257 @@
+        }
+        
+        ptrVersus = nullptr;
+        if (element->FindElement("versus")) {
+            string property_string = element->FindElementValue("versus");
+            ptrVersus = new FGParameterValue(property_string, PropertyManager);
+            double setVersus = ptrVersus->GetValue();
+            if (ptrVersus->IsConstant()) {
+                if (setVersus >= 0.5) {
+                    versus = 1;
+                } else if (setVersus <= -0.5) {
+                    versus = -1;
+                }
+            }
+        } else versus = 0;
+        
+        ptrBias = nullptr;
+        if (element->FindElement("bias")) {
+            string property_string = element->FindElementValue("bias");
+            ptrBias = new FGParameterValue(property_string, PropertyManager);
+            bias = ptrBias->GetValue();
+        } else bias = 0;
+        
+        if (element->FindElement("module")) {
+            module = element->FindElementValueAsNumber("module");
+            if (module < 0) module = 1.0;
+        }
+        if (element->FindElement("hysteresis")) {
+            hysteresis = element->FindElementValueAsNumber("hysteresis");
+            if (hysteresis < 0) hysteresis = 0.0;
+        }
+        
+        if (element->FindElement("lag")) {
+            double aLag = element->FindElementValueAsNumber("lag");
+            if (aLag > 0.0) {
+                lag = aLag;
+                double denom = 2.00 + dt*lag;
+                ca = dt * lag / denom;
+                cb = (2.00 - dt * lag) / denom;
+            }
+        }
+        
+        if (element->FindElement("rate")) {
+            double aRate = element->FindElementValueAsNumber("rate");
+            if (aRate <= 0 or aRate > 1.0) aRate = 0.5;
+            rate = aRate;
+        }
+        
+        if (element->FindElement("gain")) gain = element->FindElementValueAsNumber("gain");
+        
+        FGFCSComponent::bind();
+        
+        Debug(0);
+    }
+    
+    // %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    FGLinear_Actuator::~FGLinear_Actuator()
+    {
+        Debug(1);
+    }
+    
+    // %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    bool FGLinear_Actuator::Run(void )
+    {
+        
+        if (isSetProperty) set = abs(setProperty) >= 0.5;
+        if (isResetProperty) reset = abs(resetProperty) >= 0.5;
+        
+        if (reset) {
+            input_mem = 0.0;
+            countSpin = 0;
+            direction = 0;
+            Output = 0.0;
+            input_lost = 0.0;
+        } else {
+            if (set) {
+                Input = InputNodes[0]->getDoubleValue() - input_lost;
+                double input_delta = Input - input_mem;
+                if (abs(input_delta) >= hysteresis) {
+                    if (ptrVersus && !ptrVersus->IsConstant()) {
```

Comment:
- Do not test `ptrVersus ` existence because it is always initialized (at least with `ptrVersus = FGRealValue(0.0);`
- Do not test `ptrVersus` constantness, it's obfuscating the code and you will most likely not save a measurable number of cycles by doing that.